staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -328,6 +328,12 @@ func checkName(obj runtime.Object, name, namespace string, namer ScopeNamer) err
 //   interfaces
 func setObjectSelfLink(ctx context.Context, obj runtime.Object, req *http.Request, namer ScopeNamer) error {
 	if utilfeature.DefaultFeatureGate.Enabled(features.RemoveSelfLink) {
+		// Ensure that for empty lists we don't return <nil> items.
```

Comment:
FYI - this was clearly a bug previously. Fortunately it was caught by an existing integration test.

Reply:
can you comment why the fix is here in `setObjectSelfLink`? Doesn't sound like this func's task is to set an items slice to non-nil.

Reply:
For consistency - before this change (i.e. before adding this branch with return) - setting that list to non-nil was happening before in this function too. I mean - it still does - see line 370.

I agree it would be better to move it elsewhere, but that's orthogonal to this PR.